master/internal/db/postgres_trial.go

```diff
@@ -414,7 +414,7 @@ WHERE id = $1
 
 // setTrialBestValidation sets `public.trials.best_validation_id` to the `id` of the row in
 // `public.validations` corresponding to the trial's best validation.
-func setTrialBestValidation(tx *sqlx.Tx, id int) error {
+func setTrialBestValidation(tx *sqlx.Tx, trialId int, trialRunId int, stepsCompleted int) error {
```

Comment:
this diff is not strictly necessary for the PR, but i noticed the query could be slow sometimes.

this change is basically just to avoid pulling all the validations for a trial and sorting them every time we want to add a new validation.

by union the latest with the current best.  basically just an ugly version of:

```
    new_max = max(old_max, new_value)
```